frame/contracts/src/gas.rs

```diff
@@ -181,17 +182,40 @@ impl<T: Config> GasMeter<T> {
 		self.gas_left = self.gas_left.saturating_add(adjustment).min(self.gas_limit);
 	}
 
+	/// Update the `ref_time` component of the `gas_left` to bring it into line with
+	/// the Wasm engine gas meter. Passed `consumed` value is scaled by multiplying it by the
+	/// weight of a basic operation, as such an operation in wasmi engine costs 1.
+	///
+	/// This is used for gas synchronizations with the engine both on enter and on exit
+	/// of every host function.
+	///
+	/// Returns the updated `ref_time` value for the gas left in the meter.
+	/// Normally this would never fail, as engine should fail first when out of gas.
+	pub fn sync_reftime(&mut self, consumed: u64) -> Result<u64, DispatchError> {
```

Comment:
Name sounds as if this is a setter but it actually consumes fuel. Also let's make the result type safe.

```suggestion
	pub fn charge_fuel(&mut self, wasmi_fuel: u64) -> Result<Weight, DispatchError> {
```

Reply:
it turns out that we have to have `sync_fuel()` func (see below)